packages/react-variant-view/src/component/pathogenicity/ClinVarInterpretation.tsx

```diff
@@ -1,93 +1,66 @@
 import { computed } from 'mobx';
 import { ClinVar } from 'genome-nexus-ts-api-client';
 import { Button } from 'react-bootstrap';
 import { DefaultTooltip } from 'cbioportal-frontend-commons';
+import {
+    getClinVarInterpretation,
+    getRcvCountMap,
+    getRcvData,
+} from 'react-mutation-mapper';
 
-interface IClinVarInterpretationProps {
+export interface IClinVarInterpretationProps {
     clinVar: ClinVar | undefined;
 }
 
-enum ClinVarOrigin {
-    GERMLINE = 'germline',
-    SOMATIC = 'somatic',
+function clinVarTooltip() {
+    return (
+        <DefaultTooltip
+            placement="top"
+            overlay={<span>ClinVar Interpretation</span>}
+        >
+            <Button bsStyle="link" className="btn-sm p-0">
+                ClinVar Interpretation
+            </Button>
+        </DefaultTooltip>
+    );
 }
 
-type rcvDisplayData = {
-    name: string;
-    content: string;
-};
-
 @observer
 class ClinVarInterpretation extends React.Component<
     IClinVarInterpretationProps
 > {
     @computed get rcvCountMap() {
         if (this.props.clinVar) {
-            const clinVar = this.props.clinVar;
-            const filteredRcv = clinVar.rcv.filter(
-                rcv =>
-                    rcv.origin === ClinVarOrigin.GERMLINE ||
-                    rcv.origin === ClinVarOrigin.SOMATIC
-            );
-            const rcvMap = _.groupBy(filteredRcv, d => d.origin);
-            const rcvCountMap = _.mapValues(rcvMap, rcvs =>
-                _.countBy(rcvs, rcv => rcv.clinicalSignificance)
-            );
-            return rcvCountMap;
+            return getRcvCountMap(this.props.clinVar);
         }
+
         return undefined;
     }
 
     @computed get rcvDisplayData() {
         if (this.rcvCountMap) {
-            return _.map(
-                Object.entries(this.rcvCountMap),
-                ([origin, clinicalSignificanceCountMap]) => {
-                    return {
-                        name: origin,
-                        content: _.join(
-                            _.map(
-                                Object.entries(clinicalSignificanceCountMap),
-                                ([clinicalSignificance, count]) => {
-                                    return `${clinicalSignificance} (${count} evidences)`;
-                                }
-                            ),
-                            ', '
-                        ),
-                    };
-                }
-            );
+            return getRcvData(this.rcvCountMap);
         }
         return undefined;
     }
 
     @computed get clinVarContent() {
-        // first map by origin, then count evidence number for each origin group
         return this.rcvDisplayData ? (
             <div className={featureTableStyle['feature-table-layout']}>
                 <div className={featureTableStyle['data-source']}>
-                    {this.clinVarTooltip()}
+                    {clinVarTooltip()}
```

Comment:
this is where we can then just use normal component synax if we just make it a function componet.